pkg/webservice/handlers.go

```diff
@@ -869,3 +891,90 @@ func getGroupResourceUsage(w http.ResponseWriter, r *http.Request) {
 		buildJSONErrorResponse(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+func getEventsByType(w http.ResponseWriter, r *http.Request) {
```

Comment:
I think this may be problematic. Since we filter after retrieving a page of results, there's no way for the client to know what ID to request next (a filtered page will look the same as the last page with < {pageSize} entries). It may be better to have the clients do their own filtering. Alternatively, we need to provide metadata in the response of each ID along with its associated message. Otherwise it's impossible for a client to reliably interpret the results.

Reply:
Also, this could catastrophically fail if all results were filtered out. Imagine if ID 1-1000 are all of the wrong type; the client would request a page of 1000 entries starting with entry 1, and interpret the empty result as end-of-stream. On the next request, it would re-request the same ID and get the same results, leading to deadlock.

Reply:
What about providing both startId and endId? Or perhaps the number of filtered entries.

Reply:
That's better than nothing but doesn't give any ability to correlate messages with the ID they were generated with. Also, if a client is interested in more than one filtered data stream, they now need to query for all of them individually resulting in more work server-side. Why not just filter on the client? Data transfer is cheaper than CPU time.

Reply:
I removed the filtering.